docs/operate/reference/services/discovery/_index.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Discovery Service"
+linkTitle: "Discovery"
+description: "Use a discovery service to discover available resources on a machine."
+type: docs
+weight: 50
+no_list: true
+#icon: true
+#images: ["/services/icons/discovery.svg"]
+tags: ["discovery", "services"]
+no_service: true
+date: "2025-03-03"
+# updated: ""  # When the content was last entirely checked
+# SMEs: John N.
+modulescript: true
+---
+
+A discovery service allows you to return a list of physical hardware available on a machine, and suggest configurations for those components to integrate the hardware into the machine.
+If you are [creating a modular resource](/operate/get-started/other-hardware/) that depends on other resources that are discoverable in a systematic way, you can create a discovery service as part of your module to discover those resources.
```

Comment:
I think also here we want to be more specific than resources. If it's hardware then I'd say hardware resources and if it's viam resources then I think add the glossary entry so there's no room for misinterpretation. Still not sure whether it's meant just for hardware or for hardware or configured components/services

Reply:
I mean, it is just an API. So someone could use it differently. I believe the idea for it was to surface hardware, but you could implement it differently--just have to return a list of resource configs.